proto/ord_schema.proto

```diff
@@ -514,15 +515,59 @@ message ReactionWorkup {
   enum WorkupType {
     UNSPECIFIED = 0;
     CUSTOM = 1;
-    THERMAL_QUENCH = 2;
-    SOLVENT_QUENCH = 3;
-    EXTRACTION = 4;
-    FLASH_CHROMATOGRAPHY = 5;
+    // Addition (quench, dilution, extraction solvent, etc.)
+    // Specify composition/amount in "components".
+    ADDITION = 2;
+    // Change of temperature.
+    // Specify conditions in "temperature".
+    TEMPERATURE = 3;
+    // Concentration step, often using a rotovap.
+    CONCENTRATION = 4;
+    // Liquid extractions are often preceded by Additions. If there
+    // are multiple distinct additions prior to an extraction, it is 
+    // assumed that the kept phases are pooled.
+    // Specify which phase to keep in "keep_phase".
+    EXTRACTION = 5;
+    // Filtration (can keep solid or filtrate).
+    // Specify which phase to keep in "keep phase".
+    FILTRATION = 6;
+    // Washing a solid or liquid, keeping the original phase.
+    // Specify "components" of rinse. Rinses performed in 
+    // multiple stages should be given multiple workup steps
+    WASH = 7;
+    // Dried under vacuum.
+    DRY_IN_VACUUM = 8;
+    // Dried with chemical additive.
+    // Specify chemical additive in "components".
+    DRY_WITH_MATERIAL = 9;
+    // Purification by flash chromatography. 
+    FLASH_CHROMATOGRAPHY = 10;
+    // Purification by other prep chromatography.
+    OTHER_CHROMATOGRAPHY = 11;
+    // Scavenging step (e.g., pass through alumina pad)
+    // Specify any material additives in "components".
+    SCAVENGING = 12;
+    // Waiting step. Specify "duration".
+    WAIT = 13;
+    // Mixing step. Specify "stirring"
+    STIRRING = 14;
+    CRYSTALLIZATION = 15;
+    // pH adjustments should specify "components" to define
+    // species used as well as "ph" for target ph
+    PH_ADJUST = 16;
+    // Redissolution considered to be a special form of addition.
+    // Specify "components"
+    DISSOLUTION = 17;
   }
   WorkupType type = 1;
   string details = 2;
-  TemperatureConditions thermal_quench = 3;
-  Compound solvent_quench = 4;
+  Time duration = 3;
+  repeated Compound components = 4;
```

Comment:
optional: Do we want to add a oneof here somewhere?

Reply:
(We'll definitely have some logic in the validator to check for the appropriate fields, but my rule of thumb is to push as much validation into the schema as possible.)

Reply:
I'm still wary of oneofs, but I understand your reasoning. I was trying to maintain as much flexibility as possible since I think we don't yet have a complete understanding of how workups will need to be defined. I could see components+target_ph and components+keep_phase used simultaneously. Temperature, stirring, keep_phase, and target_ph should theoretically be mutually exclusive though. 

My inclination is to leave it as-is, but keep issue #17 open so we don't forget to come back to this